src/graphql/context.ts

```diff
@@ -8,13 +9,35 @@ export interface IContext {
 }
 
 export class Context implements IContext {
-	private accountId: string;
+	private signedMessage: string;
 
-	constructor(accountId: string) {
-		this.accountId = accountId;
+	/**This constructor will recieve singed messages for any transactions
+	 * and return the address which the message came from to be used as
+	 * context variable in other scripts.
+	 */
+	constructor(signedMessage: string) {
+		this.signedMessage = signedMessage;
 	}
 
 	public getEtheriumAccountId(): string {
-		return this.accountId;
+		return this.findAccount();
+	}
+
+	private findAccount() {
+		const web3 = new Web3(
+			new Web3.providers.HttpProvider('http://localhost:7545')
+		);
+
+		let etheriumAccountId: string = '';
+
+		try {
+			etheriumAccountId = web3.eth.accounts.recover(
+				'Auth',
+				this.signedMessage
+			);
+		} catch {
+			return '';

```

Comment:
Return null or undefined here because resolvers rely on the fact that this would be a 'falsy' value. I am unsure if empty string is falsy.

Reply:
just checked, empty string does count as a falsy, should i still change or is empty string good

Reply:
Lets still change to null or undefined. Reason:
```
if(context == null) { //works if context is null or undefined, but not if ''
    return null
}
```

Reply:
okay